core/src/main/java/org/apache/iceberg/MetadataTableUtils.java

```diff
@@ -32,9 +32,16 @@ public static boolean hasMetadataTableName(TableIdentifier identifier) {
   public static Table createMetadataTableInstance(Table table, MetadataTableType type) {
     if (table instanceof BaseTable) {
       return createMetadataTableInstance(table, metadataTableName(table.name(), type), type);
+    } else if (table instanceof HasTableOperations) {
```

Comment:
Another option is to make the check for SerializableTable only here to make impact/risk smaller.  

But I suppose it's pretty unlikely anyone is using this API on another version of HasTableOperations and expecting a failure.

Reply:
I would keep it as it is. If the incoming table implements `HasTableOperations` we can create a metadata table above that, so IMHO it should be ok